lib/model/SwitchingChannel.js

```diff
@@ -0,0 +1,61 @@
+module.exports = class SwitchingChannel {
+  constructor(key, triggerChannel) {
+    this._key = key;
+    this.triggerChannel = triggerChannel; // calls the setter
+  }
+
+  set triggerChannel(triggerChannel) {
+    this._triggerChannel = triggerChannel;
+    this._cache = {};
+  }
+
+
+  get key() {
+    return this._key;
+  }
+
+  get triggerChannelKey() {
+    return this._triggerChannel.key;
+  }
+
+  get triggerCapabilities() {
```

Comment:
We could add another utility feature: Ranges that each possible switched channel is used in.